pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/column/PhysicalColumnIndexContainer.java

```diff
@@ -281,17 +281,18 @@ public static BaseImmutableDictionary loadDictionary(PinotDataBuffer dictionaryB
   }
 
   private static ForwardIndexReader<?> loadRawForwardIndex(PinotDataBuffer forwardIndexBuffer, DataType dataType) {
-    switch (dataType.getStoredType()) {
+    DataType storedType = dataType.getStoredType();
+    switch (storedType) {
       case INT:
       case LONG:
       case FLOAT:
       case DOUBLE:
-        return new FixedByteChunkSVForwardIndexReader(forwardIndexBuffer, dataType);
+        return new FixedByteChunkSVForwardIndexReader(forwardIndexBuffer, storedType);
       case STRING:
       case BYTES:
-        return new VarByteChunkSVForwardIndexReader(forwardIndexBuffer, dataType);
+        return new VarByteChunkSVForwardIndexReader(forwardIndexBuffer, storedType);
       default:
-        throw new IllegalStateException("Illegal data type for raw forward index: " + dataType);
+        throw new IllegalStateException("Illegal data type for raw forward index: " + storedType);
```

Comment:
Recommend putting `dataType` here to log the actual type instead of the stored type

Reply:
Done.